python/plugins/per-project-location/location/per_project_location.py

```diff
@@ -0,0 +1,77 @@
+import os
+import sys
+import platform
+import functools
+import tempfile
+import logging
+
+import ftrack_api
+from ftrack_api.entity.location import Location
+from ftrack_api.accessor.disk import DiskAccessor
+from ftrack_api.structure.standard import StandardStructure
+
+
+logger = logging.getLogger(
+    'per_project_location'
+)
+
+
+class PerProjectLocation(Location):
+    platform = platform.system()
+
+    def _get_project_root(self, component):
+        target_project = component['version']['asset']['parent']['project']
```

Comment:
I would use a projection here to avoid hitting cached values:

`target_project = session.query("select disk,root from...")`

Reply:
fair call.
I'll update.

Reply:
@henriknorin-ftrack 

guess you mean this ? 

`        root_folder = self.session.query(
            f'select disk, root from Project where id is "{target_project["id"]}"'
        ).first()
`
        
        
the other data are just the project, which is unlikely to change.

Reply:
nice, thanks @seebergs !